packages/flutter/lib/src/widgets/routes.dart

```diff
@@ -665,10 +665,13 @@ abstract class ModalRoute<T> extends TransitionRoute<T> with LocalHistoryRoute<T
   /// `dependenciesChanged` method:
   ///
   /// ```dart
+  /// ModalRoute<dynamic> _route;
+  ///
   /// @override
   /// void dependenciesChanged() {
   ///  super.dependenciesChanged();
-  ///  ModalRoute.of(context).addScopedWillPopCallback(askTheUserIfTheyAreSure);
+  ///  _route = ModalRoute.of(context);
+  ///  _route.addScopedWillPopCallback(askTheUserIfTheyAreSure);
```

Comment:
Good point. In this case _route might be null before or after the `ModalRoute.of(context)` line.  Similarly, _route could be null at dispose time. 
```
void dependenciesChanged() {
  super.dependenciesChanged();
  _route?.removeScopedWillPopCallback(askTheUserIfTheyAreSure);
  _route = ModalRoute.of(context);
  _route?.addScopedWillPopCallback(askTheUserIfTheyAreSure);
}
```